### PR TITLE
Fixes and improvements

### PR DESCRIPTION
**Pull request by jriwanek**
1) Update for curse API changes
2) Adds support for 'include' type dependencies (Removed from PR as it's only supported on the site for now)
3) Adds support for java 9 as curse provides it as an option

### CODE REVIEW

**Comment by UpcraftLP** [comment]
(closes #16)

File: src/main/groovy/com/matthewprenger/cursegradle/CurseGradlePlugin.groovy [hunk 1]
```diff
@@ -14,7 +14,7 @@ class CurseGradlePlugin implements Plugin<Project> {
     static final String EXTENSION_NAME = 'curseforge'
 
     static final Set<String> VALID_RELEASE_TYPES = ['alpha', 'beta', 'release']
-    static final Set<String> VALID_RELATIONS = ['requiredLibrary', 'embeddedLibrary', 'optionalLibrary', 'tool', 'incompatible']
+    static final Set<String> VALID_RELATIONS = ['requiredDependency', 'embeddedDependency', 'optionalDependency', 'tool', 'incompatible', 'include']
```
Comment (ZeldoKavira) [inline src/main/groovy/com/matthewprenger/cursegradle/CurseGradlePlugin.groovy]: Valid relations are ['embeddedLibrary', 'incompatible', 'optionalDependency', 'requiredDependency', 'tool']

File: src/main/groovy/com/matthewprenger/cursegradle/CurseRelation.groovy [hunk 2]
```diff
@@ -6,25 +6,40 @@ import static com.matthewprenger.cursegradle.Util.check
...
-        addRelation("requiredLibrary", slugIn)
+        addRelation("requiredDependency", slugIn)
     }
+    void requiredDependency(String slugIn) {
+        addRelation("requiredDependency", slugIn)
+    }
+    @Deprecated
     void embeddedLibrary(String slugIn) {
-        addRelation("embeddedLibrary", slugIn)
+        addRelation("embeddedDependency", slugIn)
```
Comment (ZeldoKavira) [inline src/main/groovy/com/matthewprenger/cursegradle/CurseRelation.groovy]: embeddedLibrary stayed

File: src/main/groovy/com/matthewprenger/cursegradle/CurseRelation.groovy [hunk 3]
```diff
@@ -6,25 +6,40 @@ import static com.matthewprenger.cursegradle.Util.check
...
+    }
+    void embeddedDependency(String slugIn) {
+        addRelation("embeddedDependency", slugIn)
```
Comment (ZeldoKavira) [inline src/main/groovy/com/matthewprenger/cursegradle/CurseRelation.groovy]: Does not exist

File: src/main/groovy/com/matthewprenger/cursegradle/CurseRelation.groovy [hunk 4]
```diff
@@ -6,25 +6,40 @@ import static com.matthewprenger.cursegradle.Util.check
...
+    void optionalDependency(String slugIn) {
+        addRelation("optionalDependency", slugIn)
     }
     void tool(String slugIn) {
         addRelation("tool", slugIn)
     }
     void incompatible(String slugIn) {
         addRelation("incompatible", slugIn)
     }
+    void include(String slugIn) {
```
Comment (ZeldoKavira) [inline src/main/groovy/com/matthewprenger/cursegradle/CurseRelation.groovy]: While we do support include on the site I believe the API will reject this one

**Review by ZeldoKavira — CHANGES_REQUESTED** [review]

File: src/main/groovy/com/matthewprenger/cursegradle/CurseRelation.groovy (hunk 4 above)
Comment (jriwanek) [reply src/main/groovy/com/matthewprenger/cursegradle/CurseRelation.groovy]: @ZeldoKavira Can I get a full confirmation on whether the API will actually throw up on that one or not please? if it does, I'll comment it out till the API does actually support it.

Also, in Related, on site and therefore I assume also the API, Minecraft 1.13.1, Java 10-12 arn't available choices, while Java 11 and 12 are still EA, 11 be stable soon.

Comment (matthewprenger) [reply src/main/groovy/com/matthewprenger/cursegradle/CurseRelation.groovy]: I just ran a test build using the include relation and the API refused it with the following error:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':curseforge223797'.
> [CurseForge 223797] Error Code 1002: Error in field `metadata`:
  * Value "include" is not defined in enum. Path 'relations.projects[0].type', line 1, position 287.
```

**Comment by matthewprenger** [comment]
Everything else looks good and Curse has since updated their API (https://authors.curseforge.com/docs/api) to reflect the changes. If you can get the `include` relation removed I'll get this merged and released.

**Comment by jriwanek** [comment]
@matthewprenger Updated as requested, enjoy

**Pull request merged**